db/migrate/20240729084803_create_user_api_key_clients.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class CreateUserApiKeyClients < ActiveRecord::Migration[7.1]
+  def up
+    create_table :user_api_key_clients do |t|
+      t.string :client_id, null: false
+      t.string :application_name, null: false
+      t.string :public_key
+      t.string :auth_redirect
+
+      t.timestamps
+    end
+
+    add_index :user_api_key_clients, %i[client_id], unique: true
+
+    execute "INSERT INTO user_api_key_clients (client_id, application_name, created_at, updated_at)
+             SELECT client_id, application_name, created_at, updated_at
+             FROM user_api_keys"
+
+    add_reference :user_api_keys, :user_api_key_client, index: true, foreign_key: true
```

Comment:
Is it possible to rewrite this to avoid using `add_reference`? We have an unenforced (yet), rule against `add_reference` in Rubocop: 

https://github.com/discourse/discourse/blob/ccf86d9b9b52df2ffdf35d36c72ae7feac4100f0/.rubocop.yml#L4-L8 

`add_reference` is a bit too magic and we are wary of using it especially in cases where a lot of data is being migrated.

Reply:
Done! https://github.com/discourse/discourse/pull/28119/commits/00fce82bdb8106ef1e67cead7cdb65cb96eccf81

Reply:
I didn't add a foreign key, as looking through your migrations I see that there hasn't been a foreign key added since 2019/20. Is that also a current preference? (to not use foreign keys).

Reply:
Thanks! Yes, we generally avoid using foreign keys, too.